plugins/com.google.cloud.tools.eclipse.util/templates/appengine/web.xml.ftl

```diff
@@ -1,9 +1,15 @@
+<#if version == "2.5">
+  <#assign namespace = "http://java.sun.com/xml/ns/javaee"
+           schema = "web-app_2_5.xsd">
+<#elseif version == "3.1">
+  <#assign namespace = "http://xmlns.jcp.org/xml/ns/javaee"
+           schema = "web-app_3_1.xsd">
+</#if>
 <?xml version="1.0" encoding="utf-8"?>
-<web-app xmlns="http://java.sun.com/xml/ns/javaee"
+<web-app xmlns="${namespace}"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xmlns:web="http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd"
-         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd" 
-         version="2.5">
+         xsi:schemaLocation="${namespace} ${namespace}/${schema}"
```

Comment:
${namespace}/${schema} should be a single variable, schemaUrl

There's no rule that the namespace URL is part of the schema location.